File: app/src/main/java/i/am/shiro/amai/widget/TapDetector.java
package i.am.shiro.amai.widget;

import android.content.Context;
import android.view.GestureDetector;
import android.view.MotionEvent;

import com.annimon.stream.function.Consumer;

final class TapDetector {

    private final GestureDetector gestureDetector;

    TapDetector(Context context, Consumer<MotionEvent> tapListener) {
        gestureDetector = new GestureDetector(context, new GestureDetector.SimpleOnGestureListener() {
            @Override
            public boolean onSingleTapUp(MotionEvent e) {
                tapListener.accept(e);
                return true;
            }
        });
    }

    boolean onTouchEvent(MotionEvent ev) {
        return gestureDetector.onTouchEvent(ev);
    }
}

File: app/src/main/java/i/am/shiro/amai/widget/TapDetector.kt
package i.am.shiro.amai.widget

import android.content.Context
import android.view.GestureDetector
import android.view.GestureDetector.SimpleOnGestureListener
import android.view.MotionEvent
import com.annimon.stream.function.Consumer

internal class TapDetector(context: Context?, tapListener: Consumer<MotionEvent?>) {

    private val gestureDetector = GestureDetector(context, object : SimpleOnGestureListener() {
        override fun onSingleTapUp(e: MotionEvent): Boolean {
            tapListener.accept(e)
            return true
        }
    })

    fun onTouchEvent(ev: MotionEvent?) {
        gestureDetector.onTouchEvent(ev)
    }
}